File: src/main/java/com/example/demo/mapper/BlacklistIpMapper.java
package com.example.demo.mapper;

import com.example.demo.model.BlacklistIp;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.util.List;

@Mapper
public interface BlacklistIpMapper {
    int insert(BlacklistIp ip);

    boolean existsByIp(@Param("ipAddress") String ipAddress);

    List<BlacklistIp> getAll();

    int upsertIncrementTimes(@Param("ipAddress") String ipAddress);

    int markDeletedById(@Param("id") Long id);
}

File: src/main/kotlin/com/example/demo/mapper/BlacklistIpMapper.kt
package com.example.demo.mapper

import com.example.demo.model.BlacklistIp
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface BlacklistIpMapper {
    fun insert(ip: BlacklistIp): Int
    fun existsByIp(@Param("ipAddress") ipAddress: String): Boolean
    fun getAll(): List<BlacklistIp>
    fun upsertIncrementTimes(@Param("ipAddress") ipAddress: String): Int
    fun markDeletedById(@Param("id") id: Long): Int
}
